src/apps/lisk/sign_tx.py

```diff
@@ -0,0 +1,126 @@
+from trezor import wire, ui
+from trezor.messages.LiskSignedTx import LiskSignedTx
+from trezor.messages.LiskTransactionType import *
+from trezor.messages import FailureType
+from trezor.utils import HashWriter
+from apps.lisk.layout import *
+from ubinascii import unhexlify, hexlify
+
+
+async def lisk_sign_tx(ctx, msg):
+    from trezor.crypto.hashlib import sha256
+
+    public_key, seckey = await _get_keys(ctx, msg)
+    transaction = update_raw_tx(msg.transaction, public_key)
+
+    # throw ValueError if transaction has not valid structure
+    try:
+        await require_confirm_by_type(ctx, transaction)
```

Comment:
also this try/catch should maybe extend over `_get_transaction_bytes` too, because `require_confirm_by_type` does not use all relevant fields. If that's the intention. See below.